src/test/java/org/primeframework/mvc/cors/CORSFilterTest.java

```diff
@@ -217,6 +255,107 @@ public void options() throws Exception {
     assertEquals(response.headers().firstValue("Vary").orElse(null), "Origin");
   }
 
+  @Test(dataProvider = "get_included_path_pattern")
+  public void options_excluded_uri_checker(String path, boolean expectCorsAllow) throws Exception {
+    // arrange
+    corsConfiguration = new CORSConfiguration().withAllowCredentials(true)
+                                               .withAllowedMethods(HTTPMethod.GET, HTTPMethod.POST, HTTPMethod.HEAD, HTTPMethod.OPTIONS, HTTPMethod.PUT, HTTPMethod.DELETE)
+                                               .withAllowedHeaders("Accept", "Access-Control-Request-Headers", "Access-Control-Request-Method", "Authorization", "Content-Type", "Last-Modified", "Origin", "X-FusionAuth-TenantId", "X-Requested-With")
+                                               .withAllowedOrigins(URI.create("*"))
+                                               .withExcludeUriPredicate(u -> u.startsWith("/foo"))
+                                               .withExposedHeaders("Access-Control-Allow-Origin", "Access-Control-Allow-Credentials")
+                                               .withPreflightMaxAgeInSeconds(1800);
+
+    // act
+    HttpClient client = HttpClient.newBuilder().followRedirects(Redirect.NEVER).priority(256).build();
+    HttpResponse<Void> response = client.send(
+        HttpRequest.newBuilder(URI.create("http://localhost:9080" + path))
+                   .method("OPTIONS", BodyPublishers.noBody())
+                   .header("Access-Control-Request-Method", "GET")
+                   .header("Origin", "https://jackinthebox.com")
+                   .build(),
+        BodyHandlers.discarding()
+    );
+
+    // assert
+    var expectedStatus = expectCorsAllow ? 204 : 403;
+    assertEquals(response.statusCode(), expectedStatus);
```

Comment:
Can't we use the normal test simulator for these requests, or does using this REST client do something different here?

Reply:
No good reason, just following the pattern that was already in the test class (same for header assertions)

Reply:
updating now